### PR TITLE
feat: add request() and response()

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
- add common functions

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Review by lonnieezell — APPROVED** [review]
Ha! Now my docs on Shield work lol. 

Thanks for adding these!

**Review by paulbalandan — APPROVED** [review]

**Comment by iRedds** [comment]
Is it possible to know the reason for implementing these features?
In one of our discussions, your point was that the IncomingRequest class object should only be accessed in the controller.
But this implementation is contrary to your own beliefs.

The controller also has access to the Response class.

File: tests/system/CommonFunctionsTest.php [hunk 1]
```diff
@@ -137,6 +138,31 @@ public function testRedirectDefault()
         $this->assertInstanceOf(RedirectResponse::class, redirect());
     }
 
+    public function testRequestIncomingRequest()
+    {
+        Services::createRequest(new App());
+
+        $request = request();
+
+        $this->assertInstanceOf(IncomingRequest::class, $request);
```
Comment (MGatner) [inline tests/system/CommonFunctionsTest.php:147]: You should be able to assert that they reference the same object. 
```suggestion
        $this->assertInstanceOf(IncomingRequest::class, $request);
        $this->assertSame(Services::request(), $request);
```

**Review by MGatner — APPROVED** [review]
These are terribly convenient. I do want to point out that we now have a global helper function wrapping a service locator wrapping a mediator (for request type). No more layers than this, please!

**Review by MGatner — APPROVED** [review]

**Comment by kenjis** [comment]
Added the docs.

**Comment by kenjis** [comment]
@iRedds See https://github.com/codeigniter4/shield/pull/417#pullrequestreview-1098743114
Yes, I don't think these are good parts.
Because basically we don't need to use these in our apps.

But we have already `Services::request()` and `service('request')`,
and it seems many devs like global functions.
`request()` is a bit better than `service('request')`, because it can tell return type.

**Comment by iRedds** [comment]
I hope this is a balanced decision.
Because too much flexibility gives too many headaches to maintain a framework.

**Pull request merged**